phpseclib/Net/SSH2.php

```diff
@@ -4152,4 +4164,42 @@ function setWindowSize($columns = 80, $rows = 24)
         $this->windowColumns = $columns;
         $this->windowRows = $rows;
     }
+
+    /**
+     * @return string
+     */
+    public function __toString()
+    {
+        return $this->getResourceId();
+    }
+
+    /**
+     * @return string
+     */
+    public function getResourceId()
+    {
+        return '{' . spl_object_hash($this) . '}';
```

Comment:
From the line he added in `_parse_path`...

```php
if (preg_match('/^{[a-z0-9]+}$/i', $host)) {
```
If he doesn't do the curly brackets than `localhost` would match the regex and it'd try to re-use a non-existant `SSH2::$connections['localhost']` connection.

Here's how you'd do it with libssh2:

```php
<?php
$ssh2 = ssh2_connect('domain.tld');
ssh2_auth_password($ssh2, 'user', 'pass');

$sftp = ssh2_sftp($ssh2);

$arr = scandir("ssh2.sftp://$sftp/home/user/public_html");

print_r($arr);
```
If you do `echo "ssh2.sftp://$sftp/home/user/public_html"` you get `ssh2.sftp://Resource id #4/home/user/public_html`.

idk I think {} works..